pkg/handler/collector/oci/oci.go

```diff
@@ -182,109 +210,286 @@ func (o *ociCollector) getTagsAndFetch(ctx context.Context, repo string, tags []
 // Note: fetchOCIArtifacts currently does not re-check if a new sbom or attestation get reuploaded during polling with the same image digest.
 // A workaround for this would be to run the collector again with a specific tag without polling and ingest like normal
 func (o *ociCollector) fetchOCIArtifacts(ctx context.Context, repo string, rc *regclient.RegClient, image ref.Ref, docChannel chan<- *processor.Document) error {
-	logger := logging.FromContext(ctx)
-
 	// attempt to request only the headers, avoids Docker Hub rate limits
 	m, err := rc.ManifestHead(ctx, image)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed retrieving manifest head: %w", err)
 	}
 
+	// check if the manifest is a manifest list
 	if m.IsList() {
-		m, err := rc.ManifestGet(ctx, image)
-		if err != nil {
+		if err := o.fetchManifestList(ctx, repo, rc, image, docChannel); err != nil {
 			return err
 		}
-		pl, _ := manifest.GetPlatformList(m)
-		for _, p := range pl {
+	}
+
+	// check for fallback artifacts
+	if err := o.fetchFallbackArtifacts(ctx, repo, rc, image, m, docChannel); err != nil {
+		return err
+	}
+
+	// check for referrer artifacts
+	if err := o.fetchReferrerArtifacts(ctx, repo, rc, image, docChannel); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// fetchManifestList fetches the manifest list for the given image and fetches all the platform manifests in parallel.
+// It then fetches the artifacts for each platform and sends them to the docChannel.
+// It returns an error if any error occurs during the process.
+func (o *ociCollector) fetchManifestList(ctx context.Context, repo string, rc *regclient.RegClient, image ref.Ref, docChannel chan<- *processor.Document) error {
+	logger := logging.FromContext(ctx)
+
+	m, err := rc.ManifestGet(ctx, image)
+	if err != nil {
+		return fmt.Errorf("failed retrieving manifest: %w", err)
+	}
+
+	pl, _ := manifest.GetPlatformList(m)
```

Comment:
should check the error here. This was already there in the previous version of the code...not sure why that was never checked.